tests/generic_relations_regress/tests.py

```diff
@@ -332,3 +333,16 @@ def test_generic_reverse_relation_exclude_filter(self):
         self.assertSequenceEqual(qs, [link2])
         qs = Link.objects.exclude(places__name="Test Place 1")
         self.assertSequenceEqual(qs, [link2])
+
+    def test_ticket_34816_generic_fk_crash(self):
+        """
+        Ensure pk type change is handled when using GenericForeignKey.
+        """
```

Comment:
To add more explanation, docstrings should state the expected behavior and omit prefixes like "Ensure that" since all tests ensure things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style). Maybe
```suggestion
        """Primary key is not checked if the content type doesn't match."""
```